hypno/api.py

```diff
@@ -49,3 +50,34 @@ def inject_py(pid: int, python_code: AnyStr, permissions=0o644) -> None:
     finally:
         if path is not None and path.exists():
             path.unlink()
+
+def inject_py_script(pid: int, python_script: Path, permissions=0o644) -> None:
```

Comment:
Much of the logic here is duplicated from above. I think it might be better to keep it as one function, `inject_py`, and check `isinstance(python_code, Path)` to determine the logic. Also, I think that instead of modifying `injection.c`, it would be easier to just read the given file here, and use the contents as code.

Reply:
If a large script is provided then it will overflow. Also relative imports will not work as expected either presumably. To me it made sense to use Python API's file parsing instead of shoehorning this in?

Reply:
Yeah I agree about the advantages, but ideally I'd want to enocunter all possible errors before the injection, so they happen in python and not in C in the target process. I also wonder permission-wise, if it's better to use the injector process permissions or the injectee's permissions (and fs, btw, see #18). I will hopefully have more time to think about it later.